runtime/gc_stats.c

```diff
@@ -101,13 +102,20 @@ void caml_orphan_alloc_stats(caml_domain_state *domain) {
   caml_plat_unlock(&orphan_lock);
 }
 
-
 /* The "sampled stats" of a domain are a recent copy of its
    domain-local stats, accessed without synchronization and only
    updated ("sampled") during stop-the-world events -- each minor
    collection, major cycle (which potentially includes compaction),
    all of these events could happen during domain termination. */
-static struct gc_stats sampled_gc_stats[Max_domains];
+static struct gc_stats* sampled_gc_stats;
+
+void caml_init_gc_stats (uintnat max_domains)
+{
+  sampled_gc_stats =
+    caml_stat_alloc(max_domains * sizeof(struct gc_stats));
+  if (sampled_gc_stats == NULL)
+    caml_fatal_error("Failed to allocate sampled_gc_stats");
```

Comment:
I think that you meant `caml_stat_alloc_noexc`, as `caml_stat_alloc` fails with Out_of_memory instead of returning NULL. (I'm not sure if the hand-written init message is worth it, no strong opinion.) The other calls in the patch are all to `_noexc` so they are fine.

Reply:
Done in https://github.com/ocaml/ocaml/pull/13272/commits/263be361baad53f6ef36f8774ac8b5d9b1572525.